// Copyright 2020 The 9nFL Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package jdfl;

//
// coordinator and proxy
//

service PairService {
  rpc RegisterUUID(Request) returns (Status);
  rpc GetPairInfo(Request) returns (PairInfoResponse);
}

message Status {
  // 0:  success
  // others: fail
  uint32 status = 1; 
  string err_msg = 2;
}

message Request {
  // AppID_Service_index
  repeated string uuid = 1;
  // ip:port
  string ip_port = 2;
}

message ServicePair {
  string local_uuid = 1;
  string remote_uuid = 2;
}

message PairInfoResponse {
  Status status = 1;
  repeated ServicePair service_map = 2;
}

